x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/AuthorizationEngine.java

```diff
@@ -349,7 +349,7 @@ public boolean isAuditable() {
          * Returns additional context about an authorization failure, if {@link #isGranted()} is false.
          */
         @Nullable
-        public String getFailureContext() {
+        public String getFailureContext(Predicate<String> isRestrictedIndex) {
```

Comment:
I want to replace this predicate with a new `RestrictedIndices` class (possibly also subsuming the existing `RestrictedIndexNames` class) but that would have impacted more code than this PR needed to touch, so I'll do it as a followup.